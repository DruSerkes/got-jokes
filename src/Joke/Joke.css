.Joke {
  margin-top: 3rem;
}