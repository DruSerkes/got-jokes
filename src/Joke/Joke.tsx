import * as React from 'react';
import { jokeData } from '../types';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Spinner from 'react-bootstrap/Spinner'

interface JokeProps {
    isLoading: boolean;
    joke?: jokeData;
}

export const Joke = ({ joke, isLoading }: JokeProps) => {
    return (
        <Container className='Joke'>
            {!joke && <Spinner animation='grow' />}
            {
                !isLoading && joke?.type === 'twopart' &&
                <>
                    <Row className='Joke-Setup my-3'>
                        <Col>
                            {joke?.setup}
                        </Col>
                    </Row>
                    <Row className='Joke-Delivery my-3'>
                        <Col>
                            {joke?.delivery}
                        </Col>
                    </Row>
                </>
            }
            {
                !isLoading && joke?.type === 'single' &&
                <Row className='my-3'>
                    <Col>
                        {joke?.joke}
                    </Col>
                </Row>
            }
        </Container>
    )
}