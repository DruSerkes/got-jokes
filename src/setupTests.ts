// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';

export const testJoke = {
    "error": false,
    "category": 'dark',
    "type": 'single',
    "joke": 'test joke',
    "flags": {
        "nsfw": false,
        "religious": false,
        "political": false,
        "racist": false,
        "sexist": false,
        "explicit": false
    },
    "id": 13,
    "safe": true,
    "lang": 'en',
    'setup': 'test setup',
    'delivery': 'test delivery'
};

export const setJoke = jest.fn(() => console.log('setting joke'));
export const setViewingFavorites = jest.fn(() => console.log('viewing favorites'));
export const setFavorites = jest.fn(() => console.log('setting favorites'));
export const clearFavorites = jest.fn(() => console.log('clearing favorites'));
export const saveFavoriteJoke = jest.fn(() => console.log('saving favorite joke'));