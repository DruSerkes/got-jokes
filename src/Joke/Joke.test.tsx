import * as React from 'react';
import { render, screen } from '@testing-library/react';
import { Joke } from './Joke';
import { testJoke } from '../setupTests';

describe('Joke tests', () => {

    it('renders without breaking and matches snapshot', () => {
        const { asFragment } = render(<Joke isLoading={false} joke={testJoke} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders nothing if isLoading is true', () => {
        render(<Joke isLoading={true} joke={testJoke} />);
        const theJoke = screen.queryByText(testJoke.joke);
        expect(theJoke).not.toBeInTheDocument();
    });

    it('displays the joke on the screen', () => {
        render(<Joke isLoading={false} joke={testJoke} />);
        const theJoke = screen.getByText(testJoke.joke);
        expect(theJoke).toBeInTheDocument();
    });

    it('displays the setup and the delivery for joke with type of twopart', () => {
        testJoke.type = 'twopart';
        render(<Joke isLoading={false} joke={testJoke} />);
        const theSetup = screen.getByText(testJoke.setup);
        const theDelivery = screen.getByText(testJoke.delivery);
        expect(theSetup).toBeInTheDocument();
        expect(theDelivery).toBeInTheDocument();
    });
})