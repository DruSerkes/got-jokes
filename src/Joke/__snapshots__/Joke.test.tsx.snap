// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Joke tests renders without breaking and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Joke container"
  >
    <div
      class="my-3 row"
    >
      <div
        class="col"
      >
        test joke
      </div>
    </div>
  </div>
</DocumentFragment>
`;
